### DISCUSSION TITLE
Does GitHub Packages support Nuget Symbol Packages (.snupkg)?

### DISCUSSION THREAD

**Discussion by cjoergensen** [community/community#38678] [Packages] [answered]
Hi

Does GitHub packages supports Nuget Symbol Packages (.snupkg)?

https://learn.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg

**Comment by ossanna16** [comment]
Hi there @cjoergensen and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Reply by DavidThompsonWG** [reply]
@cjoergensen Did you ever get a reponse to this or figure out if they are supported. I am trying to find the same info, but not having much joy. Thanks, David

**Reply by cjoergensen** [reply]
@DavidThompsonWG nope, no response so far

**Reply by dmstrat** [reply]
I too am awaiting a response here as it seems to be failing if you try: 

dotnet nuget push "/directory_of_both_files/" --api-key as;lkfjasdl;kfj --source "github nuget package reference name" 

It explicitly rejects the secondary upload of the *.snupkg file with a 'version conflict'.  This exactly matches to the stackoverflow post: https://stackoverflow.com/questions/52142074/dotnet-nuget-push-doesnt-seem-to-be-pushing-the-symbols  

I've read the updated Github Packages documentation: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry as well, but no  mention of the snupkg uploads.  

Would like to have something in the searchable documentation to know one way or the other as I would like to keep these together for promotion to release once verified.

**Comment by trichardsonPDS** [comment, marked as answer]
It's been a while since I looked into this but after some digging I discovered that GitHub does not support Nuget Symbols Packages (forget exactly where I saw that as it's been a while).

I was however able to work around this by using Source Link (https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink).  The trick is to embed the symbols file (.pdb) in the DLL itself instead of trying to use a separate file/package.  I was able to accomplish this with the following two lines of code in the project file:

`<DebugType>Embedded</DebugType>`
`<IncludeSymbols>true</IncludeSymbols>`

You'll also need to include a package reference to Source Link.  You can find more details here: https://github.com/dotnet/sourcelink/blob/main/README.md

Once I had everything configured and I pulled down the updated Nuget package, debugging just started working.  

Hope that helps....

**Reply by dmstrat** [reply]
Thank you for adding this information.  The embed seemed like the next step.  You took it even further which is greatly appreciated.
